test/fstapp_SUITE.erl

```diff
@@ -4,26 +4,38 @@
 
 -compile(export_all).
 
-init_per_suite(Config) ->
-	application:ensure_all_started(fstapp),
+init_per_testcase(_, Config) ->
+	true = register(fstapp_SUITE_process, self()),
+	_ = application:ensure_all_started(fstapp),
 	Config.
 
-end_per_suite(Config) ->
-	application:stop(fstapp),
-	Config.
+end_per_testcase(_, _Config) ->
+	application:stop(fstapp).
 
 all() ->
-	[my_test_case].
+        [metrics_test, frequency_test].
+
+metrics_test(_Config) ->
 
-my_test_case(_Config) ->
+        receive
+                Metrics ->
+                	{cpu, _} = lists:keyfind(cpu, 1, Metrics),
+                        {ostype, _} = lists:keyfind(ostype, 1, Metrics),
+                        {proc, _} = lists:keyfind(proc, 1, Metrics),
+                        {disk, _} = lists:keyfind(disk, 1, Metrics)
+	after
+		10000 ->
+			{error, error_reason}
```

Comment:
we have to do something here which warns the user the test didn't pass. Returning this tuple is not going to help